### PR TITLE
bind: factor out RecursiveUnmount

### PR DESCRIPTION
**Pull request by nalind**
Factor out logic for recursively unmounting our bind mounts, and make use of the parent/child relationship between mountpoints to decide which order to use, rather than sorting based on their mountpoint paths.

### CODE REVIEW

**Comment by rhatdan** [comment]
Seems rather brutal.  But LGTM

File: bind/mount.go [hunk 1]
```diff
@@ -201,3 +198,97 @@ func leaveBindMountAlone(mount specs.Mount) bool {
 	}
 	return false
 }
+
+// RecursiveUnmount unmounts the given mountpoints and anything that's hanging
+// off of them, rather aggressively.  If a mountpoint also appears in the
+// mountpointsToRemove slice, the mountpoints are removed after they are
+// unmounted.
+func RecursiveUnmount(mountpoint string, mountpointsToRemove []string) error {
```
Comment (TomSweeneyRedHat) [inline bind/mount.go]: nit on the name, not really recursive is it?  FullUnmount, UmountAll, HardUmount, or something similar instead?

Comment (nalind) [reply bind/mount.go]: The function isn't recursive, true.  The name mirrors that of the similar function in the `github.com/containers/storage/pkg/mount` package, which does something similar, but which selects the order in which it unmounts things by sorting the directory paths of the mountpoints, as we did previously, and which doesn't detach busy filesystems.  If you have a preference for a different name, I'm happy to change it.

Comment (TomSweeneyRedHat) [reply bind/mount.go]: Would UnmountMountpoints work for you?

Comment (nalind) [reply bind/mount.go]: Sure.  Making the change.

**Comment by nalind** [comment]
Rebased.

**Comment by TomSweeneyRedHat** [comment]
LGTM, thanks for the funtion name change to keep my OCD at bay!  :^)

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**